tests/test_new_return_types.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the new return types.
+"""
+import pytest
+
+import numpy as np
+import pennylane as qml
+
+herm = np.diag([1, 2, 3, 4])
+probs_data = [
+    (None, [0]),
+    (None, [0, 1]),
+    (qml.PauliZ(0), None),
+    (qml.Hermitian(herm, wires=[1, 0]), None),
+]
+
+wires = [2, 3, 4]
+
+
+class TestSingleReturnExecute:
+    """Test that single measurements return behavior does not change."""
+
+    @pytest.mark.parametrize("wires", wires)
+    def test_state_default(self, wires):
```

Comment:
For all of these single return tests, it would also be good to test if a tuple of one element is returned from the QNode, i.e.
```python
@qml.qnode(dev)
def circuit():
    qml.Hadamard(0)
    return [qml.expval(qml.PauliZ(0))]
```

Reply:
The custom return type is adressed in the third PR: https://github.com/PennyLaneAI/pennylane/pull/2860